File: Java/ToyLanguageGUI_exam_final/src/utils/OutList.java
package utils;

import java.util.ArrayList;
import java.util.List;

public class OutList<T> implements IOutList<T> {
    private List<T> l;
    public OutList(){
        l = new ArrayList<>();
    }


    public void addMsg(T m){
        l.add(m);
    }

    @Override
    public String toString(){
        StringBuffer buff = new StringBuffer();
        for(T e : l)
            buff.append(""+e+"\n");
        return buff.toString();
    }

    @Override
    public Iterable<T> getAll(){
        return l;
    }
}

File: ToyInterpretor/ToyInterpretor/Utils/OutList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToyInterpretor.Utils
{
    class OutList<T> : IOutList<T>
    {
        private List<T> l;
        public OutList()
        {
            l = new List<T>();
        }


        public void addMsg(T m)
        {
            l.Add(m);
        }

        
        public override String ToString()
        {
            StringBuilder buff = new StringBuilder();
            foreach (T e in l)
                buff.Append("" + e + "\n");
            return buff.ToString();
        }

        
        public IEnumerable<T> getAll()
        {
            return l;
        }
    }
}
